src/envoy/http/authn/authenticator_base.cc

```diff
@@ -68,7 +73,21 @@ bool AuthenticatorBase::validateX509(const iaapi::MutualTls& mtls,
 bool AuthenticatorBase::validateJwt(const iaapi::Jwt& jwt, Payload* payload) {
   std::string jwt_payload;
   if (filter_context()->getJwtPayload(jwt.issuer(), &jwt_payload)) {
-    return AuthnUtils::ProcessJwtPayload(jwt_payload, payload->mutable_jwt());
+    std::string payload_to_process = jwt_payload;
+    std::string original_payload;
+    bool found = false;
+    for (auto h : jwt.jwt_headers()) {
```

Comment:
For readability, make a helper function (say IsAPTokenIssuer) for this.

Reply:
Done.